runtime/zeitgeist/src/lib.rs

```diff
@@ -96,17 +96,21 @@ pub struct IsCallable;
 
 // Currently disables Court, Rikiddo and creation of markets using Court or SimpleDisputes
 // dispute mechanism.
+// Disallow GlobalDisputes until we use Court. GlobalDisputes with authorized is not allowed.
```

Comment:
You mean it should be completely removed from both runtimes `battery-station` and `zeitgeist` for now? Couldn't I just remove the `Call` from `GlobalDisputes: zrml_global_disputes::{Event<T>, Pallet, Storage} = 56,`? Problem when I would remove the pallet from the  runtimes: I can't pass GlobalDisputes in prediction-market pallet then, so I need to delete the logic from the prediction market then.

Reply:
The functionality of global disputes must be completely excluded from the Zeitgeist (mainnet) runtime until it passed a test cycle on Battery Station. This means that ideally the runtime does not include it yet, but afaik not exposing the call enum should be sufficient. In addition to that, any part of the system beyond global disputes should not use the functionality of global disputes, e.g. prediction markets should not use global disputes (on mainnet). On testnet though, all the functionality should be available. This might result in having to migrate the implementation of the `Config trait` from the common runtime into each respective runtime.
To disable the global disputes API in the prediction markets pallet, it might be sufficient to provide a dummy implementation that does enough to maintain proper functioning without the global disputes pallet, in some cases this is nothing (like for the [liquidity mining api](https://github.com/zeitgeistpm/zeitgeist/blob/294c66c4d070171fbe4389fbb63b1608e7df3ba8/runtime/common/src/lib.rs#L826-L845))

Reply:
That seems like a huge head struggle, because it's complicated to execute. Ok, we have the drawback, that users see the call on the main-net, that's a nice sneak peek 😄. I totally see your point, but there are a few things to consider with that approach:
We could use a feature flag and mark each global dispute occurrence in the pm-pallet and common runtime.
We have the following things.
- [`start_global_dispute` extrinsic](https://github.com/zeitgeistpm/zeitgeist/blob/6cea4e3feae5c7e7d6ee5d993aa0617a75162849/zrml/prediction-markets/src/lib.rs#L447-L514)
- [`resolution_manager`](https://github.com/zeitgeistpm/zeitgeist/blob/6cea4e3feae5c7e7d6ee5d993aa0617a75162849/zrml/prediction-markets/src/lib.rs#L2367-L2374) 
- [`Config types`](https://github.com/zeitgeistpm/zeitgeist/blob/6cea4e3feae5c7e7d6ee5d993aa0617a75162849/zrml/prediction-markets/src/lib.rs#L1208-L1217) 
- [`Error enum types`](https://github.com/zeitgeistpm/zeitgeist/blob/6cea4e3feae5c7e7d6ee5d993aa0617a75162849/zrml/prediction-markets/src/lib.rs#L1290-L1293)
- [`Event enum type`](https://github.com/zeitgeistpm/zeitgeist/blob/6cea4e3feae5c7e7d6ee5d993aa0617a75162849/zrml/prediction-markets/src/lib.rs#L1342-L1343)
- [`on_resolution` winner determination ](https://github.com/zeitgeistpm/zeitgeist/blob/6cea4e3feae5c7e7d6ee5d993aa0617a75162849/zrml/prediction-markets/src/lib.rs#L1985-L1987)

Maybe we can call the feature `with-global-disputes` and pass the argument to the `battery-station`, but not to the `main-net`. We could mark all occurrences of global disputes in the common runtime with that feature flag too.

Then we can delete all occurrences of `global-disputes` in `/runtime/zeitgeist/Cargo.toml`, or can't we. I need to check that.

Reply:
923c2812b6099b00fb10222365551bb8a60896bb

Currently I found to way to allow an whole extrinsic when a feature is activated.